### ISSUE TITLE
agent: Sending query error" error="query exceeds limit of 1024 bytes"

### ISSUE THREAD

**Issue by careys7** [victorcoder/dkron#249]
When running commands that have a large output, we get the below error in the log and both the agent and server crash. 

In our case we were testing with the help output of a shell command, but also had the issue when our application crashed with a stack trace as foreground output. 

```
[dkron-1]2017-03-30T05:03:07.763435764Z time="2017-03-30T05:03:07Z" level=debug msg="agent: Sending query" job_name=secondjob json="{\"execution\":{\"job_name\":\"secondjob\",\"started_at\":\"2017-03-30T05:03:04.860203661Z\",\"finished_at\":\"2017-03-30T05:03:05.216618567Z\",\"output\":\"CkZhdGFsIGVycm9yOiBVbmNhdWdodCBTeW1mb255XENvbXBvbmVudFxEZWJ1Z1xFeGNlcHRpb25cQ2xhc3NOb3RGb3VuZEV4Y2VwdGlvbjogQXR0ZW1wdGVkIHRvIGxvYWQgY2xhc3MgIlNlbnNpb0dlbmVyYXRvckJ1bmRsZSIgZnJvbSBuYW1lc3BhY2UgIlNlbnNpb1xCdW5kbGVcR2VuZXJhdG9yQnVuZGxlIi4KRGlkIHlvdSBmb3JnZXQgYSAidXNlIiBzdGF0ZW1lbnQgZm9yIGFub3RoZXIgbmFtZXNwYWNlPyBpbiAvb3B0L2FwcC92ZW5kb3IvYmFsYW5jZS9pbnRlZ3JhdG9yLWZyYW1ld29yay9zcmMvQ29tcG9uZW50L0tlcm5lbC9LZXJuZWwucGhwOjQzClN0YWNrIHRyYWNlOgojMCAvb3B0L2FwcC9hcHAvQXBwS2VybmVsLnBocCg4KTogSW50ZWdyYXRvclxGcmFtZXdvcmtcQ29tcG9uZW50XEtlcm5lbFxLZXJuZWwtPnJlZ2lzdGVyQnVuZGxlcygpCiMxIC9vcHQvYXBwL3ZlbmRvci9zeW1mb255L3N5bWZvbnkvc3JjL1N5bWZvbnkvQ29tcG9uZW50L0h0dHBLZXJuZWwvS2VybmVsLnBocCg0MDYpOiBBcHBLZXJuZWwtPnJlZ2lzdGVyQnVuZGxlcygpCiMyIC9vcHQvYXBwL3ZlbmRvci9zeW1mb255L3N5bWZvbnkvc3JjL1N5bWZvbnkvQ29tcG9uZW50L0h0dHBLZXJuZWwvS2VybmVsLnBocCgxMTMpOiBTeW1mb255XENvbXBvbmVudFxIdHRwS2VybmVsXEtlcm5lbC0+aW5pdGlhbGl6ZUJ1bmRsZXMoKQojMyAvb3B0L2FwcC92ZW5kb3Ivc3ltZm9ueS9zeW1mb255L3NyYy9TeW1mb255L0J1bmRsZS9GcmFtZXdvcmtCdW5kbGUvQ29uc29sZS9BcHBsaWNhdGlvbi5waHAoNjgpOiBTeW1mb255XENvbXBvbmVudFxIdHRwS2VybmVsXEtlcm5lbC0+Ym9vdCgpCiM0IC9vcHQvYXBwL3ZlbmRvci9zeW1mb255L3N5bWZvbnkvc3JjL1N5bWZvbnkvQ29tcG9uZW50L0NvbnNvbGUvQXBwbGljYXRpb24ucGhwKDEyMCk6IFN5bWZvbnlcQnVuZGxlXEZyYW1ld29ya0J1bmRsZVxDb25zb2xlXEFwcGxpY2F0aW9uLT5kb1J1bihPYmplY3QoU3ltZm9ueVxDb21wb25lbnRcQ29uc29sZVxJbnB1dFxBcmd2SW5wdXQpLCBPYmplY3QoU3ltZm9ueVxDIGluIC9vcHQvYXBwL3ZlbmRvci9iYWxhbmNlL2ludGVncmF0b3ItZnJhbWV3b3JrL3NyYy9Db21wb25lbnQvS2VybmVsL0tlcm5lbC5waHAgb24gbGluZSA0Mwo=\",\"node_name\":\"scheduler-agent-executor-1\",\"group\":1490850184029231348,\"attempt\":2},\"rpc_addr\":\"10.7.0.6:6868\"}" node=dkron-1 query="run:job" 
[dkron-1]2017-03-30T05:03:07.763759017Z time="2017-03-30T05:03:07Z" level=fatal msg="agent: Sending query error" error="query exceeds limit of 1024 bytes" node=dkron-1 query="run:job" 
```

We have worked around this using head to limit the output of the job:

```
/opt/app/my/command | head -c512
```

Is this a known limitation or user error? I noticed #110 which seems similar.

**Comment by victorcoder** [comment]
Hi @careys7 what Dkron version are you using? This limitation was gone since 0.9.0 https://github.com/victorcoder/dkron/blob/master/CHANGELOG.md#090-2016-08-24

**Comment by careys7** [comment]
Hi @victorcoder the agent and server are both using 0.9.3:

```
Dkron v0.9.3                                                                                                                                                             
Agent Protocol: 4 (Understands back to: 2)                                                                                                                               
```

The server is using the official Docker image, agent is using an image based off `php:7.1-alpine`, installing dkron from Github.

**Comment by careys7** [comment]
Is there any logs I can provide to help with this?

**Comment by victorcoder** [comment]
This weird, I'll do some checks and I'll ping you back

**Comment by victorcoder** [comment]
This is really weird @careys7 I did some tests with 249Kb responses and it works as expected.

Said that, it's a bug in retries code, a fix will be released in the upcoming 0.9.5

**Comment by careys7** [comment]
@victorcoder thanks for checking. 

This is the relevant parts from our Docker Cloud stack file if you can see anything wrong?

```yml
consul:
  command: agent -server -bootstrap -ui -client=0.0.0.0 -advertise=127.0.0.1
  hostname: consul
  image: 'consul:latest'
  ports:
    - '8300:8300'
    - '8500:8500'
  tags:
    - nodecluster-name=scheduler

dkron:
  command: 'agent -server -backend=consul -backend-machine=consul:8500 -join=dkron:8946 -advertise=dkron -log-level=debug'
  image: 'dkron/dkron:latest'
  ports:
    - '8080:8080'
  tags:
    - nodecluster-name=scheduler

scheduler-agent-executor:
  image: 'repo/image-name:latest'
  tags:
    - nodecluster-name=scheduler
```

The scheduler-agent-exector Dockerfile is as follows (we have another image between php:7.1-alpine that adds some dependencies which is omitted below):

```
FROM php:7.1-alpine

ENV			DKRON_VERSION 0.9.3

RUN			set -x \
			&& buildDeps='bash ca-certificates openssl' \
			&& apk add --update $buildDeps \
			&& rm -rf /var/cache/apk/* \
			&& mkdir -p /opt/local/dkron \
			&& wget -O dkron.tar.gz https://github.com/victorcoder/dkron/releases/download/v${DKRON_VERSION}/dkron_${DKRON_VERSION}_linux_amd64.tar.gz \
			&& tar -xzf dkron.tar.gz \
			&& mv dkron_${DKRON_VERSION}_linux_amd64/* /opt/local/dkron \
			&& rm dkron.tar.gz \
			&& rm -rf dkron_${DKRON_VERSION}_linux_amd64
			
EXPOSE      6868 8946		

ADD			scripts/run.sh /run.sh
RUN			chmod 755 /run.sh	

ENV 		SHELL	/bin/bash
WORKDIR		/opt/local/dkron

CMD 	    ["/run.sh"]
```

run.sh is as follows:

```sh
#!/bin/sh

/opt/local/dkron/dkron agent -tag="role=executor" -advertise=$HOSTNAME -join=dkron:8946 -log-level=debug
```
